src/v3/src/components/AutoStepper/AutoStepper.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { useEffect } from 'preact/hooks';
+
+import { useStepperContext } from '../../contexts';
+import { AutoStepperElement, UISchemaElementComponent } from '../../types';
+
+const AutoStepper: UISchemaElementComponent<{
```

Comment:
Overall, this is fine and works, but it is set up for a very specific scenario. If we wanted to we could make this a little more generic by creating a `callback` property in the AutoStepperElement interface, and using that callback in the useEffect of this component. This way, the transformers can define what this "Navigator" Component  should do, this also would fall more in line with the new name suggestion of this Component and interface. 

The callback function can accept the Stepper context as a parameter (to allow the transformer to do what it needs to do), and this component can pass the context into the callback. 

Not sure if this is overkill but I just want to avoid future duplication of this sort of thing and give future devs a path forward to do this sort of thing.

Reply:
Would love to hear your thoughts @shawyu-okta maybe this is just overkill.

Reply:
@glenfannin-okta can you give an example of how that would look? I'm wondering what the parameter to the callback would be.

Reply:
Yessir, something like this; 

Interface: 
```
export interface StepperNavigatorElement {
  type: 'StepperNavigator',
  options: {
     callback: (stepperContext: IStepperContext) => void;
  }
} 
```

Component:
```
const StepperNavigator: UISchemaElementComponent<{
  uischema: StepperNavigatorElement
}> = ({ uischema }) => {
  const stepperContext = useStepperContext();
  const {
    options: {
      callback,
    },
  } = uischema;

  useEffect(() => {
    callback(stepperContext);
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return null;
};

export default StepperNavigator;
```

Then finally, the transformer:
```
...
if (challengeMethod === CHALLENGE_METHOD.APP_LINK
      && prevTransaction?.nextStep?.name === IDX_STEP.IDENTIFY) {
    const stepperNavigatorElement: StepperNavigatorElement = {
      type: 'StepperNavigator',
      options: {
        callback: (stepperContext: IStepperContext) => {
           const { setStepIndex } = stepperContext;
           setTimeout(() => setStepIndex!(1), FASTPASS_FALLBACK_SPINNER_TIMEOUT);
        },
      },
    };

    uischema.elements.push({
      type: UISchemaLayoutType.STEPPER,
      elements: [
        {
          type: UISchemaLayoutType.VERTICAL,
          elements: [
            stepperNavigatorElement,
            spinnerElement,
            cancelLink,
          ],
        } as UISchemaLayout,
        {
          type: UISchemaLayoutType.VERTICAL,
          elements: [
            descriptionElement,
            openOktaVerifyButton,
            cancelLink,
          ],
        } as UISchemaLayout],
    });

    return formBag;
  }
...
```

Reply:
My thinking with making this component was that it may be more flexible since it could be placed in any of the `elements` indexes lets say if you didn't want it to do the auto step on the first page but when the user lands on a different page/index.  You could also place multiple in different indexes if you wanted to have multiple transitions.  But that is only for possible future use not guaranteed that it will be needed so your method if having the callback in the `AutoStepperElement` would work too.

Reply:
I see, yea the above modifications would enable the same, the only real difference is that this approach delegates to the transformer to tell the component _how_ it should navigate between steps. 

For example, the current approach only allows it to navigate using a setTimeout, whereas, there could be other scenarios where it doesn't want to use a setTimeout, but instead maybe navigate based on a data change or something else.

But its up to you guys, if you feel like this is overkill for now then I totally understand.

Reply:
I generally favor an approach of simpler encapsulated behavior as long as it doesn't prevent us from expanding to more generalized behavior in the future. That would favor the existing approach for now, although if we run into other use-cases we should implement as @glenfannin-okta suggested. I don't think it would be too much work to update the transformers that use this component if we do rework this component to take a callback in the future.

Reply:
ahh I see the call back is in the `StepperNavigator` component so it will have the same flexibility with the added flexibility of being able to navigate on a custom condition.  I do like that and am cool with implementing it that way.

Reply:
@shawyu-okta  true it wouldn't be too difficult to update it in the future so that sounds good to me.  What do you think of the name should it remain the same or change it to 'StepperNavigator'?

Reply:
Yeah im gonna see if I can implement glens approach shouldnt be too much work